examples/turmoil-provider/src/lib.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#![cfg(test)]
+
+use s2n_quic::{
+    client::Connect,
+    provider::{event, io},
+    Client, Server,
+};
+use std::net::SocketAddr;
+use turmoil::{lookup, Builder};
+
+type Result<T = (), E = Box<dyn std::error::Error>> = core::result::Result<T, E>;
```

Comment:
You can take [turmoil's](https://github.com/tokio-rs/turmoil/blob/main/src/error.rs#L8) if you'd like.